### PR TITLE
sysocli-file: update jdupes

### LINKED ISSUE #5505: jdupes (from synocli file tools) no dedupe support
### Is this a new Bug?

- [X] I checkd that the bug hasn't been reported before

### Package Name

SynoCli File Tools

### Package Version

2.7-18

### Device Model

ds218+

### Device Architecture

x86_64

### Firmware Version

DSM 7.1.1-42962 Update 1

### What happened?

```
jdupes --dedupe -1 -r /volume2/...
This program was built without dedupe support
```

Took a look at the Makefile ( https://github.com/SynoCommunity/spksrc/blob/master/cross/jdupes/Makefile ) and original source (https://github.com/jbruchon/jdupes/blob/master/jdupes.c#L2129). It would seem source needs to be compiled with ```-DENABLE_DEDUPE=1``` for this to work (https://github.com/jbruchon/jdupes/blob/8f5b06109b44a9e4316f9445da3044590a6c63e2/Makefile#L17).

Not familiar enough with the build tooling to know (or check) whether just adding it to the current ```jdupes_install``` will suffice; I reckon one would need to override ```MAKEFLAGS``` ? 

### Reproduction steps

```bash
1. jdupes --dedupe -1 -r /volume*/
```

### Install Log

```log
If you must...

2022/11/27 23:14:28	install synocli-file 2.7-18 Begin preinst
2022/11/27 23:14:29	Begin reload_inst_variables
2022/11/27 23:14:29	End reload_inst_variables
2022/11/27 23:14:29	Begin initialize_variables
2022/11/27 23:14:29	End initialize_variables
2022/11/27 23:14:29	===> Step preinst. USER= GROUP= SHARE_PATH=
2022/11/27 23:14:30	install synocli-file 2.7-18 End preinst ret=[0]
2022/11/27 23:14:33	install synocli-file 2.7-18 Begin /bin/mkdir -p /volume3/@appstore/synocli-file -m 755
2022/11/27 23:14:33	install synocli-file 2.7-18 End /bin/mkdir -p /volume3/@appstore/synocli-file -m 755 ret=[0]
2022/11/27 23:14:33	install synocli-file 2.7-18 Begin /bin/rm -rf /volume3/@appstore/synocli-file
2022/11/27 23:14:33	install synocli-file 2.7-18 End /bin/rm -rf /volume3/@appstore/synocli-file ret=[0]
2022/11/27 23:14:33	install synocli-file 2.7-18 Begin /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/package /volume3/@appstore/synocli-file
2022/11/27 23:14:35	install synocli-file 2.7-18 End /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/package /volume3/@appstore/synocli-file ret=[0]
2022/11/27 23:14:35	install synocli-file 2.7-18 Begin /bin/rm -rf /var/packages/synocli-file
2022/11/27 23:14:35	install synocli-file 2.7-18 End /bin/rm -rf /var/packages/synocli-file ret=[0]
2022/11/27 23:14:35	install synocli-file 2.7-18 Begin /bin/mkdir -p /var/packages/synocli-file -m 755
2022/11/27 23:14:35	install synocli-file 2.7-18 End /bin/mkdir -p /var/packages/synocli-file -m 755 ret=[0]
2022/11/27 23:14:35	install synocli-file 2.7-18 Begin /bin/touch /var/packages/synocli-file/installing
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/touch /var/packages/synocli-file/installing ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/INFO /var/packages/synocli-file/INFO
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/INFO /var/packages/synocli-file/INFO ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/rm -rf /var/packages/synocli-file/scripts
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/rm -rf /var/packages/synocli-file/scripts ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/scripts /var/packages/synocli-file/scripts
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/scripts /var/packages/synocli-file/scripts ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/rm -rf /var/packages/synocli-file/WIZARD_UIFILES
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/rm -rf /var/packages/synocli-file/WIZARD_UIFILES ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/WIZARD_UIFILES /var/packages/synocli-file/WIZARD_UIFILES
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/WIZARD_UIFILES /var/packages/synocli-file/WIZARD_UIFILES ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/rm -rf /var/packages/synocli-file/conf
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/rm -rf /var/packages/synocli-file/conf ret=[0]
2022/11/27 23:14:36	install synocli-file 2.7-18 Begin /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/conf /var/packages/synocli-file/conf
2022/11/27 23:14:36	install synocli-file 2.7-18 End /bin/mv -f /volume3/@tmp/synopkg/install.poWW3N/conf /var/packages/synocli-file/conf ret=[0]
2022/11/27 23:15:18	install synocli-file 2.7-18 Begin postinst
2022/11/27 23:15:18	Begin reload_inst_variables
2022/11/27 23:15:18	End reload_inst_variables
2022/11/27 23:15:18	Begin initialize_variables
2022/11/27 23:15:18	End initialize_variables
2022/11/27 23:15:18	===> Step postinst. USER= GROUP= SHARE_PATH=
2022/11/27 23:15:18	Begin save_wizard_variables
2022/11/27 23:15:18	End save_wizard_variables
2022/11/27 23:15:18	Begin syno_sync_var_folder
2022/11/27 23:15:18	End syno_sync_var_folder
2022/11/27 23:15:18	Begin service_postinst
2022/11/27 23:15:18	End service_postinst
2022/11/27 23:15:18	install synocli-file 2.7-18 End postinst ret=[0]
2022/11/27 23:15:22	install synocli-file 2.7-18 Begin start-stop-status start
2022/11/27 23:15:22	install synocli-file 2.7-18 End start-stop-status start ret=[0]
```
```


### Service Log

_No response_

### Other Logs

_No response_

### PR DESCRIPTION
**Pull request by hgy59**
## Description

- update jdupes to support --dedupe

Fixes #5505

## Checklist

- [x] Build rule `all-supported` completed successfully
- [x] New installation of package completed successfully
- [x] Package upgrade completed successfully (Manually install the package again)
- [ ] Package [functionality was tested](https://github.com/SynoCommunity/spksrc/wiki/Update-Policy#tests-checks)
- [ ] Any needed [documentation](https://github.com/SynoCommunity/spksrc/wiki/Create-documentation) is updated/created

### Type of change

- [x] Bug fix
- [ ] Package update

### CODE REVIEW

File: cross/jdupes/Makefile [hunk 1]
```diff
@@ -13,10 +13,8 @@ COMMENT  = A powerful duplicate file finder and an enhanced fork of fdupes.
 LICENSE  = MIT
 
 CONFIGURE_TARGET = nop
-INSTALL_TARGET = jdupes_install
+INSTALL_MAKE_OPTIONS = install DESTDIR=$(INSTALL_DIR) PREFIX=$(INSTALL_PREFIX)
 
-include ../../mk/spksrc.cross-cc.mk
+ADDITIONAL_CFLAGS = -DENABLE_DEDUPE
```
Comment (jisakiel) [inline cross/jdupes/Makefile:18]: doesnt work (tested it)

Comment (jisakiel) [reply cross/jdupes/Makefile:18]: Set instead ```COMPILE_MAKE_OPTIONS = ENABLE_DEDUPE=1``` and it does. Tested via:
```
root@(...):/spksrc/spk/synocli-file# ./work-x64-7.1/jdupes-1.21.0/jdupes --version
jdupes 1.21.0 (2022-09-03) 64-bit
Compile-time extensions: fsdedup
```

Comment (hgy59) [reply cross/jdupes/Makefile:18]: for me it works as implemented in this PR.

aarch64-6.1 package downloaded from github build action and installed on DS218 with DSM 6.2.4 shows:
```
$ jdupes --version
jdupes 1.21.0 (2022-09-03) 64-bit
Compile-time extensions: fsdedup
Copyright (C) 2015-2022 by Jody Bruchon and contributors
Forked from fdupes 1.51, (C) 1999-2014 Adrian Lopez and contributors
....
```

you can download the packages created by github build action under "checks - build - artifacts".

Comment (jisakiel) [reply cross/jdupes/Makefile:18]: ah, I tried only the COMPILE_MAKE_OPTIONS without the rest of changes I suppose is why. My bad!

Comment (hgy59) [reply cross/jdupes/Makefile:18]: @jisakiel can you please test the dedup feature ❔ 

I am not sure, whether this version is fully self contained to dedup files. 
I have read that btrfs libraries are needed and I like to have a confirmation that no external library is missing in jdupes for dedup to work.

You can use the spk file created by github build action (see above how to download).

Comment (jisakiel) [reply cross/jdupes/Makefile:18]: It works only on appearance :(

```
jisakiel@diskstation:~$ jdupes --version | head
jdupes 1.21.0 (2022-09-03) 64-bit
Compile-time extensions: fsdedup
```

But:
```
jdupes --dedupe -1 -r /volume1/Backups/*
...
  [SRC] /volume1/Backups/(...)/(...).jpg
  -XX-> /volume1/Backups/(...)/(...)/(...)
error: Inappropriate ioctl for device (25)
```

Stracing around a bit with some sketchy static strace binary I found gets me not a whole lot of clarity more:
```
ioctl(3, _IOC(_IOC_READ|_IOC_WRITE, 0x5e, 0x36, 0x18), 0x1b0e4f0) = -1 ENOTTY (Inappropriate ioctl for device)
```

Comment (jisakiel) [reply cross/jdupes/Makefile:18]: Tested this on https://github.com/SynoCommunity/spksrc/commit/c932c4a326edad6efcf8047eeb704c27166b4006 (I don't think I can update this pr with it) and it works now:

```
jdupes --dedupe -1 -r /volume1/Backups/*
...
  [SRC] /volume1/Backups/(...).dng
  ====> /volume1/Backups/(...)/Copy(...).dng

Deduplication done (18 files processed)
```

Comment (hgy59) [reply cross/jdupes/Makefile:18]: @jisakiel thanks for testing this. I have applied the patch.

Just looking for updates of other tools in synocli-file, before building a new release...

**Pull request merged**